tests/index.test.ts

```diff
@@ -0,0 +1,89 @@
+import { ExtensibilitySdk, Message, OAuthDialogCompletedMessage } from '../src/index';
+import runtime from '../src/sdk/RuntimeContext';
+import oauthService from '../src/sdk/services/oauthService';
+
+describe('sdk tests', () => {
+  let sdk: ExtensibilitySdk;
+
+  describe.only('receiving messages', () => {
+    var messageHandler: EventListenerOrEventListenerObject;
+    let originalEventListener: (event: string, handler: EventListenerOrEventListenerObject) => void;
+    let originalOpenPopup: (redirectUri?: string, state?: { [key: string]: string }) => void;
+
+    beforeEach(async () => {
+      window.setTimeout = jest.fn().mockImplementation(() => 123) as any;
+
+      originalOpenPopup = oauthService.openPopup;
+      oauthService.openPopup = jest.fn();
+
+      originalEventListener = window.addEventListener;
+      window.addEventListener = (event: string, handler: EventListenerOrEventListenerObject): void => {
+        if (event === 'message') {
+          messageHandler = handler;
+        }
+      };
+      runtime.application = {
+        api: {
+          client: { id: '' },
+          redirectUris: [],
+          scopes: [],
+        },
+      } as any;
+      runtime.origin = 'test.outreach.io';
+      sdk = new ExtensibilitySdk();
+      sdk.init();
```

Comment:
you call this before every test. Does it have sideeffects or are we certain it's pure and will remain so?

Reply:
In order for our SDK to do anything,.init has to be called - every method checks for that before being executed so init has to be ran. Due to the previous line where a new instance of SDK is created for every test there is no side effects